foxtrot-core/src/main/java/com/flipkart/foxtrot/core/exception/FoxtrotException.java

```diff
@@ -0,0 +1,119 @@
+package com.flipkart.foxtrot.core.exception;
+
+import com.flipkart.foxtrot.common.ActionRequest;
+import com.flipkart.foxtrot.common.Table;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by rishabh.goyal on 13/12/15.
+ */
+public abstract class FoxtrotException extends Exception {
+
+    private ErrorCode code;
+
+    public FoxtrotException(ErrorCode code) {
+        this.code = code;
+    }
+
+    public FoxtrotException(ErrorCode code, String message) {
+        super(message);
+        this.code = code;
+    }
+
+    public FoxtrotException(ErrorCode code, Throwable cause) {
+        super(cause);
+        this.code = code;
+    }
+
+    public FoxtrotException(ErrorCode code, String message, Throwable cause) {
+        super(message, cause);
+        this.code = code;
+    }
+
+    public abstract Map<String, Object> toMap();
+
+    public ErrorCode getCode() {
+        return code;
+    }
+
+    public void setCode(ErrorCode code) {
+        this.code = code;
+    }
+
+    public static TableInitializationException createTableInitializationException(Table table, String message) {
```

Comment:
Won't it be better if you create another class called Exceptions and move the create functions there?